### DISCUSSION TITLE
What is the expected propagation time for a new package version?

### DISCUSSION THREAD

**Discussion by rob-nice** [microsoft/winget-pkgs#180237] [Q&A] [answered]
Patch Tuesday has happened, Microsoft has [dropped the .net SDKs 6.0.427 and 8.0.403](https://devblogs.microsoft.com/dotnet/dotnet-and-dotnet-framework-october-2024-servicing-updates/).

I can see you've merged PRs for [6.0.427](https://github.com/microsoft/winget-pkgs/pull/179906) and [8.0.403](https://github.com/microsoft/winget-pkgs/pull/179931) but neither are yet available when running
`winget upgrade --id Microsoft.DotNet.SDK.6`
or
`winget upgrade --id Microsoft.DotNet.SDK.8`

This puts us ~10 and ~8 hours post merge, which seems slow - What is the expected propagation time?  Is there any way for a consumer to see the state of a version release?

Source list as I suspect it may be relevant
```
> winget source list
Name    Argument                                      Explicit
--------------------------------------------------------------
msstore https://storeedgefd.dsx.mp.microsoft.com/v9.0 false
winget  https://cdn.winget.microsoft.com/cache        false
```

**Comment by mdanish-kh** [comment, marked as answer]
The usual propagation time is [generally within an hour](https://github.com/microsoft/winget-pkgs/blob/master/doc/FAQ.md#how-long-do-packages-take-to-be-published). You're seeing unexpected delay right now because the publish pipeline is failing at the moment. You can see the status labels from the [repo's README](https://github.com/user-attachments/assets/cab6722f-c2f1-4db6-9609-feb2573af590). 

> Is there any way for a consumer to see the state of a version release?

This is generally tracked by the https://github.com/microsoft/winget-pkgs/labels/Publish-Pipeline-Succeeded label. The bot labels a merged PR with Publish-Pipeline-Succeeded when that PR has successfully gone through the pipeline. You can see right now that the PRs you linked above don't have that label. But sometimes when the pipeline is erroring out, the maintainers may kick off a manual publish run for the whole repo, and that may not attach this label.

**Reply by rob-nice** [reply]
That explains it, thanks for the detailed response!

**Reply by mdanish-kh** [reply]
FYI: The current issue is being tracked in:

- https://github.com/microsoft/winget-pkgs/issues/180404
